console/console-init/ui/src/modules/device/components/AddCredential/AddCredential.tsx

```diff
@@ -87,11 +85,9 @@ export const AddCredential: React.FC<{}> = () => {
 
   const setSecretsInitialFormState = () => {
     const newCredentials = [...credentials];
-    const index = findIndexByProperty(
-      credentials,
-      "id",
-      activeCredentialFormId
-    );
+    const activeFormId =
+      activeCredentialFormId || newCredentials[newCredentials.length - 1].id;
+    const index = findIndexByProperty(credentials, "id", activeFormId);
```

Comment:
For `activeFormId`, I am seeing an error - undefined is not assignable to type string.
doing a `activeFormId` || `""` will probably fix it.

Reply:
not sure, but it's working fine for me. safer side I'll do some change.